### PR TITLE
Custom package repository document expansion and update

### PR DESCRIPTION
**Pull request by parlough**
Adds some documentation, cleans up existing docs, and incorporates more information from https://medium.com/dartlang/hosting-a-private-dart-package-repository-774c3c51dff9

**Staged:** https://dart-dev--pr4101-misc-custom-package-b4winpnh.web.app//tools/pub/custom-package-repositories

Fixes https://github.com/dart-lang/site-www/issues/3921

### CODE REVIEW

File: src/tools/pub/custom-package-repositories.md [hunk 1]
```diff
@@ -3,16 +3,17 @@ title: Custom package repositories
...
 The official package repository is [pub.dev]({{site.pub}}), this repository
-is by the `pub` tool by default. A package repository is identified by a
-_hosted-url_ (e.g. `https://pub.example.com/my-account/`). 
-
-Third-party package servers can be useful for hosting private packages. It is
-also common to use [git-dependencies](/tools/pub/dependencies#git-packages) for hosting
-private packages, however, the `pub` does not support resolving versions against
-a git-repository it just fetches a specific revision of the git-repository.
+is by the `dart pub` tool by default. A package repository is identified by a
+_hosted-url_ (e.g. `https://dart-packages.example.com/my-account/`).
```
Comment (sfshaza2) [inline src/tools/pub/custom-package-repositories.md:10]: Writing guidelines, we don't use "e.g" or "i.e." replace with "for example" or "such as" as appropriate.

File: src/tools/pub/custom-package-repositories.md [hunk 2]
```diff
@@ -3,16 +3,17 @@ title: Custom package repositories
...
-private packages, however, the `pub` does not support resolving versions against
-a git-repository it just fetches a specific revision of the git-repository.
+is by the `dart pub` tool by default. A package repository is identified by a
+_hosted-url_ (e.g. `https://dart-packages.example.com/my-account/`). 
+
+Third-party package servers can be useful for hosting private packages. 
+It is also common to use [git-dependencies](/tools/pub/dependencies#git-packages) 
+for hosting private packages, however, 
+the `dart pub` tool does not support resolving versions against
+a git-repository it just fetches a specific revision of the git repository.
```
Comment (sfshaza2) [inline src/tools/pub/custom-package-repositories.md:16]:
```suggestion
the `dart pub` tool doesn't support resolving versions against
a git-repository; it just fetches a specific revision of the git repository.
```

**Comment by parlough** [comment]
Thanks for the suggestions! Hoping to get back to this tomorrow and finalize the page update :)

File: src/tools/pub/custom-package-repositories.md [hunk 3]
```diff
@@ -3,16 +3,17 @@ title: Custom package repositories
 description: "How Dart's package management tool, pub, works with custom package repositories."
 ---
 
-The `pub` tool has support for third-party package repositories. A package
+The `dart pub` tool has support for third-party package repositories. A package
 repository is an HTTP server from which the `pub` client can fetch packages.
 The official package repository is [pub.dev]({{site.pub}}), this repository
```
Comment (domesticmouse) [inline src/tools/pub/custom-package-repositories.md:8]: This line feels awkward.

**Comment by parlough** [comment]
@sfshaza2 Can you take another look at these changes when you get a chance? Thanks so much :>

File: src/tools/pub/custom-package-repositories.md [hunk 4]
```diff
@@ -3,133 +3,208 @@ title: Custom package repositories
...
-## Getting a custom package repository
+### Overriding the default package repository {#default-override}
 
-The REST API for writing a custom package repository is outlined in the
-[Hosted Pub Repository Specification Version 2][repository-spec-v2.md].
+By default, `dart pub` retrieves dependencies from and publishes packages
+to the [pub.dev site]({{site.pub}})
+unless the hosted-dependency syntax
+is used to specify a custom package repository.
+However, it’s possible to override the default package repository using the
```
Comment (sfshaza2) [inline src/tools/pub/custom-package-repositories.md:188]:
```suggestion
However, you can override the default package repository using the
```

File: src/tools/pub/custom-package-repositories.md [hunk 5]
```diff
@@ -3,133 +3,208 @@ title: Custom package repositories
...
-is by the `pub` tool by default. A package repository is identified by a
-_hosted-url_ (e.g. `https://pub.example.com/my-account/`). 
-
-Third-party package servers can be useful for hosting private packages. It is
-also common to use [git-dependencies](/tools/pub/dependencies#git-packages) for hosting
-private packages, however, the `pub` does not support resolving versions against
-a git-repository it just fetches a specific revision of the git-repository.
-Hence, when many people are collaborating it is sometimes preferable to use a
-private package repository.
+The `dart pub` tool has support for third-party package repositories.
```
Comment (sfshaza2) [inline src/tools/pub/custom-package-repositories.md:6]:
```suggestion
The `dart pub` tool supports third-party package repositories.
```

File: src/tools/pub/custom-package-repositories.md [hunk 6]
```diff
@@ -3,133 +3,208 @@ title: Custom package repositories
...
+
+Sometimes a custom package repository might be useful
+for hosting private packages,
+including in some of the following scenarios:
+
+1. Sharing internal proprietary packages within an organization.
+2. Tight control of dependencies in enterprise environments.
+3. Secure environments without public internet access.
+
+It is also common to use [git-dependencies](/tools/pub/dependencies#git-packages)
```
Comment (sfshaza2) [inline src/tools/pub/custom-package-repositories.md:23]:
```suggestion
It's also common to use [git-dependencies](/tools/pub/dependencies#git-packages) 
```

File: src/tools/pub/custom-package-repositories.md [hunk 7]
```diff
@@ -3,133 +3,208 @@ title: Custom package repositories
...
+1. Sharing internal proprietary packages within an organization.
+2. Tight control of dependencies in enterprise environments.
+3. Secure environments without public internet access.
+
+It is also common to use [git-dependencies](/tools/pub/dependencies#git-packages) 
+for hosting private packages, however, 
+the `dart pub` tool doesn't support resolving versions against a git repository; 
+it just fetches a specific revision of the git repository.
+Therefore, when many people are collaborating
+it is often preferable to use a private package repository.
```
Comment (sfshaza2) [inline src/tools/pub/custom-package-repositories.md:28]:
```suggestion
it's often preferable to use a private package repository.
```

**Review by sfshaza2 — APPROVED** [review]
lgtm, after some minor edits

**Review by sfshaza2 — APPROVED** [review]
lgtm!

**Pull request merged**